### PR TITLE
[CAS-1270] Load older messages

### PR DESCRIPTION
**Pull request by ogkuzmin**
### 🎯 Goal

Deprecate usage of `ChatDomain::loadOlderMessages`

### 🛠 Implementation details

1. Create extension `ChatClient::loadOlderMessages`
2. Substitute usage of the deprecated method by a new extension

### 🧪 Testing

Try to paginate older messages and be sure it's shown

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModelTest.kt [hunk 1]
```diff
@@ -138,6 +139,7 @@ internal class MessageListViewModelTest {
     }
 
     @Test
+    @Ignore("Can not be tested until we use mockk or other way to mock static function")
```
Comment (ogkuzmin) [inline stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModelTest.kt:142]: In order to fix it we need to mock the static function. It's not possible with mockito. For that, we need to use `mockk`. If we want I can include `mockk` in this PR and fix the test

**Review by amitkma — APPROVED** [review]
LGTM

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/channel/ChannelController.kt [hunk 2]
```diff
@@ -305,76 +304,24 @@ public class ChannelController internal constructor(
...
     internal suspend fun loadOlderMessages(limit: Int = 30): Result<Channel> {
-        return runChannelQuery(
-            QueryChannelPaginationRequest(limit).apply {
-                getLoadMoreBaseMessageId(Pagination.LESS_THAN)?.let {
-                    messageFilterDirection = Pagination.LESS_THAN
-                    messageFilterValue = it
-                }
-            }.toWatchChannelRequest(domainImpl.userPresence)
-        )
+        return runChannelQuery(channelLogic.olderWatchChannelRequest(limit, null))
```
Comment (filbabic) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/channel/ChannelController.kt:309]: Maybe we could use named parameters for these functions, just so we know what the null param is specifically

If you feel like it's not important, than that's also okay! :]

**Review by samiuelson — APPROVED** [review]

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt [hunk 3]
```diff
@@ -274,6 +277,54 @@ internal class ChannelLogic(
...
+        val messageId = baseMessageId ?: getLoadMoreBaseMessageId(pagination)
+        return QueryChannelPaginationRequest(limit).apply {
+            messageId?.let {
+                messageFilterDirection = pagination
+                messageFilterValue = it
+            }
+        }.toWatchChannelRequest(chatDomainImpl.userPresence)
+    }
+
+    /** Calculates base messageId for [WatchChannelRequest] depending on [Pagination] when requesting more messages. */
```
Comment (filbabic) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt:315]: This doc piece seems to be out of place - formatting and structure wise! Could we clean it up? :]

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt [hunk 4]
```diff
@@ -147,3 +148,19 @@ public fun ChatClient.stopTyping(cid: String, parentId: String? = null): Call<Bo
         channelController.stopTyping(parentId)
     }
 }
+
+/**
+ * Loads older messages for the channel.
+ *
+ * @param cid The full channel id i.e. "messaging:123".
+ * @param messageLimit How many new messages to load.
```
Comment (filbabic) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt:156]: We should specify the return value too!

**Review by filbabic — APPROVED** [review]
Looks good to me - I suggested a few smaller docs improvements so if we can clean that up that'd be great!

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt [hunk 5]
```diff
@@ -274,6 +277,54 @@ internal class ChannelLogic(
...
+     * Returns instance of [WatchChannelRequest] to obtain newer messages of a channel.
+     *
+     * @param limit Message limit in this request.
+     * @param baseMessageId Message id of the last available message. Request will fetch messages newer than this.
+     */
+    internal fun newerWatchChannelRequest(limit: Int, baseMessageId: String?): WatchChannelRequest =
+        watchChannelRequest(Pagination.GREATER_THAN, limit, baseMessageId)
+
+    /**
+     * Creates instance of [WatchChannelRequest] according [Pagination].
```
Comment (MarinTolic) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt:299]: Perhaps "Creates instance of [WatchChannelRequest] according **to** [Pagination]." would be more accurate?

Comment (MarinTolic) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt:299]: Perhaps "according **to** `Pagination`?
Would that sound better?

**Review by MarinTolic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images) **(checked as there is no visual difference)**
- [x] Bugs validated (bugfixes) **(checked as this was a migration effort)**
- [x] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Comment by filbabic** [comment]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [x] Bugs validated (bugfixes)
- [x] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**